src/tls/Handshake.fst

```diff
@@ -537,7 +542,10 @@ let processServerHelloDone cfg n ks log msgs opt_msgs =
            | Some pk ->
 	     let valid = Signature.verify #a h pk tbs sigv in
 	     //let _ = IO.debug_print_string("tbs = " ^ (Platform.Bytes.print_bytes tbs) ^ "\n") in
-	     let _ = IO.debug_print_string("Signature validation status = " ^ (if valid then "OK" else "FAIL") ^ "\n") in
+	     let _ = 
+        if hs_debug then 
+          IO.debug_print_string("Signature validation status = " ^ (if valid then "OK" else "FAIL") ^ "\n") 
+        else false in
```

Comment:
Maybe define a `unit` returning wrapper for `IO.debug_print_string` to avoid all these `else false` branches.